### ISSUE TITLE
no static method "Lorg/mozilla/firefox/vpn/qt/VPNUtils;.initializeGlean(Landroid/content/Context;ZLjava/lang/String;)V"

### ISSUE THREAD

**Issue by data-sync-user** [mozilla-mobile/mozilla-vpn-client#6005]
Getting this error when initializing Glean on Android.

See logs excerpt:

```
[13.02.2023 11:44:49.125] (Glean) Debug: Initializing MZGlean
2023-02-13 11:44:49.128 15629-15629 System.err              org.mozilla.firefox.vpn.debug        W  java.lang.NoSuchMethodError: no static method "Lorg/mozilla/firefox/vpn/qt/VPNUtils;.initializeGlean(Landroid/content/Context;ZLjava/lang/String;)V"
2023-02-13 11:44:49.128 15629-15629 System.err              org.mozilla.firefox.vpn.debug        W  	at org.qtproject.qt.android.QtNative.runPendingCppRunnables(Native Method)
2023-02-13 11:44:49.128 15629-15629 System.err              org.mozilla.firefox.vpn.debug        W  	at org.qtproject.qt.android.QtNative$1.run(QtNative.java:128)
2023-02-13 11:44:49.128 15629-15629 System.err              org.mozilla.firefox.vpn.debug        W  	at android.os.Handler.handleCallback(Handler.java:938)
2023-02-13 11:44:49.128 15629-15629 System.err              org.mozilla.firefox.vpn.debug        W  	at android.os.Handler.dispatchMessage(Handler.java:99)
2023-02-13 11:44:49.128 15629-15629 System.err              org.mozilla.firefox.vpn.debug        W  	at android.os.Looper.loop(Looper.java:236)
2023-02-13 11:44:49.128 15629-15629 System.err              org.mozilla.firefox.vpn.debug        W  	at android.app.ActivityThread.main(ActivityThread.java:8031)
2023-02-13 11:44:49.129 15629-15629 System.err              org.mozilla.firefox.vpn.debug        W  	at java.lang.reflect.Method.invoke(Native Method)
2023-02-13 11:44:49.129 15629-15629 System.err              org.mozilla.firefox.vpn.debug        W  	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:620)
2023-02-13 11:44:49.129 15629-15629 System.err              org.mozilla.firefox.vpn.debug        W  	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1011)
```

┆Issue is synchronized with this [Jira Bug](https://mozilla-hub.atlassian.net/browse/VPN-4158)
┆Reporter: Beatriz Rizental Machado

**Comment by data-sync-user** [comment]
➤ Beatriz Rizental Machado commented:

Once this is merged, instructions for QA

Reproduction Steps

  1. Open the VPN application on Android
  2. Login in, leave data collection turned ON
  3. Use the application for a while
  4. Check that at least one ping was sent

Expected Results

  1. Data is collected

Current Results

  1. No data is collected, there was an error initializing Glean

**Comment by data-sync-user** [comment]
➤ Valentina Virlics commented:

Beatriz Rizental Machado We are blocked in checking this, using 2.14, as the last build from release branch is from 7 days ago. Could we have a new build with the latest fixes? Thank you!

**Comment by data-sync-user** [comment]
➤ Beatriz Rizental Machado commented:

Valentina Virlics it looks like the builds are available, again. Is this unblocked?

**Comment by data-sync-user** [comment]
➤ Bianca Hidecuti commented:

Verified this as fixed on Mozilla VPN 2.14.0 (2.202302162209), while using Android 11. 

Pings are sent and data is collected - see the attached ss.

!Screenshot 2023-02-20 at 10.46.22.png|width=1696,height=930!

Beatriz Rizental Machado, we noticed that the “client_info.app_build” column displays the build number in the old format. Is this expected? Thank you!

cc: Valentina Virlics

**Comment by data-sync-user** [comment]
➤ Beatriz Rizental Machado commented:

Hey Bianca Hidecuti thank you so much for checking. The app_build thing is a separate issue, I did not realize there was even an “old” or “new” format for that field, but it’s likely oversight on my part when implementing the new Glean. I’ll file a separate ticket for it.

**Comment by data-sync-user** [comment]
➤ Basti commented:

>“client_info.app_build” column displays the build number in the old format.

Oh neat! That is actually more usefull then the Timestamp format, as we can trace that file around sentry and google play 🙂
